modules/tensor_mechanics/include/materials/crystal_plasticity/CrystalPlasticityKalidindiBackstress.h

```diff
@@ -0,0 +1,93 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+/*
+   Constitutive model form:
+   Reference: Kalidindi, S. R., Bronkhorst, C. A., & Anand, L. (1992). Crystallographic texture evolution in bulk deformation processing of FCC metals. Journal of the Mechanics and Physics of Solids, 40(3), 537-569.
+
+   Plus an Armstrong-Frederick backstress term:
+   Reference: Armstrong, P. J., & Frederick, C. O. (1966). A mathematical representation of the multiaxial Bauschinger effect (Vol. 731). Berkeley, CA: Berkeley Nuclear Laboratories.
+
+   The flow rule of the shearing slip rate:
+   \dot{\gamma}^\alpha = \dot{\gamma}_o * |(\tau^\alpha - \chi^\alpha) / g^\alpha|^M * sgn(\tau^\alpha - \chi^\alpha)
+
+   Armstrong-Frederick backstress term:
+   \dot{\chi}^\alpha = h * \dot{\gamma}^\alpha - r * |\dot{\gamma}^\alpha| * \chi^\alpha
+
+   Reference: https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/CrystalPlasticityDislocationUpdate.C
+*/
+
+#pragma once
+
+#include "CrystalPlasticityKalidindiUpdate.h"
+
+class CrystalPlasticityKalidindiBackstress;
```

Comment:
Please change the class name to `CrystalPlasticityKalidindiBackstressUpdate`. We use the subfix `Update` nomenclature to indicate that these materials are a particular kind of MOOSE material class known as "discrete materials" and that these materials have to be called by a different MOOSE material class

Reply:
good

Reply:
resolved